solidity/contracts/libs/isms/MultisigIsmMetadata.sol

```diff
@@ -3,40 +3,35 @@ pragma solidity >=0.8.0;
 
 /**
  * Format of metadata:
- * [   0:  32] Merkle root
- * [  32:  36] Root index
- * [  36:  68] Origin mailbox address
- * [  68:1092] Merkle proof
- * [1092:????] Validator signatures, 65 bytes each, length == Threshold
+ * [   0:  32] Origin mailbox address
+ * [  32:  VS] Validator signatures (where VS := 32 + Threshold * 65 bytes)
+ * [  VS:????] Suffix
+ *
+ * Format of suffix:
+ * [  -38:  -1] Merkle root
+ * [   -1:   0] SuffixType.ROOT
+ *                  OR
+ * [-1061: -37] Merkle proof
+ * [  -37:  -5] Message ID
+ * [  -5:  -1] Index
+ * [   -1:   0] SuffixType.PROOF_INDEX_ID
  */
 library MultisigIsmMetadata {
-    uint256 private constant MERKLE_ROOT_OFFSET = 0;
-    uint256 private constant MERKLE_INDEX_OFFSET = 32;
-    uint256 private constant ORIGIN_MAILBOX_OFFSET = 36;
-    uint256 private constant MERKLE_PROOF_OFFSET = 68;
-    uint256 private constant SIGNATURES_OFFSET = 1092;
-    uint256 private constant SIGNATURE_LENGTH = 65;
+    uint8 private constant ORIGIN_MAILBOX_OFFSET = 0;
+    uint8 private constant SIGNATURES_OFFSET = 32;
+    uint8 private constant SIGNATURE_LENGTH = 65;
 
-    /**
-     * @notice Returns the merkle root of the signed checkpoint.
-     * @param _metadata ABI encoded Multisig ISM metadata.
-     * @return Merkle root of the signed checkpoint
-     */
-    function root(bytes calldata _metadata) internal pure returns (bytes32) {
-        return bytes32(_metadata[MERKLE_ROOT_OFFSET:MERKLE_INDEX_OFFSET]);
+    enum SuffixType {
```

Comment:
took me a pretty long time to understand why this was called root & why only the root was passed in. It wasn't immediately clear to me that one suffix type corresponded with a message ID being signed directly and another with a merkle proof

Comments explaining would help, or possibly even renaming the types to something like this imo:
```
enum MetadataType {
  SignedMessageId,
  MerkleProof
}
```
But I guess I was thinking of these types as establishing the intent of the metadata and not necessarily the contents, which I don't have a super strong take on